Lib/operator.py

```diff
@@ -307,6 +307,70 @@ def __repr__(self):
     def __reduce__(self):
         return self.__class__, self._items
 
+class itemtuplegetter:
+    """
+    Return a callable object that fetches the given items from its operand in a tuple.
+
+    If defaults is given, when called on an object where i-th `items` is not present,
+    the corresponding defaults is returned instead. If the defaults iterable is
+    shorter than subscripts iterable, the remaining subscripts have no defaults.
+    If the defaults iterable is longer than subscripts iterable, extra defaults are
+    ignored.
+
+    The returned callable has two read-only properties:
+        operator.itemtulegetter.items: a tuple containing items to fetch
+        operator.itemtulegetter.defaults: a tuple containing provided defaults
+
+    For example,
+    After f = itemtuplegetter([0, 2], defaults=(-1, -2)), f([1, 2]) evaluates to (1, -2).
+    After g = itemtuplegetter([0, 2], defaults=(-1)), f([1, 2]) resutls in an IndexError.
+    After h = itemtuplegetter([0], defaults=(-1, -2)), f([1, 2]) evaluates to (1,).
+    After i = itemtuplegetter([1, 0], defaults=(-1, -2)), f([1, 2]) evaluates to (2, 1).
+    """
+    __slots__ = ('_items', '_defaults')
+
+    def __init__(self, items, /, defaults = None):
+        self._items = tuple(items)
+        self._defaults = () if defaults is None else tuple(defaults)
```

Comment:
This will not cut it. E.g. if one sources `itertools.repeat(None)`, this goes to infinite loop.

I think logic is best to stay as:
```python
if defaults is None:
    defaults = ()
elif isinstance(defaults, (tuple, list)):
    defaults = defaults[:len(items)]
else:
    defaults = itl.islice(defaults, len(items))
if not isinstance(defaults, tuple):
    defaults = tuple(defaults)
self._defaults = defaults
```
Might need alternative to `islice` not use `itertools` if it is not a dependency. And maybe some simplification to above conditionals possible.

Reply:
Also, probably need a test for infinite defaults iterator.

Reply:
Ah, I had a version with a custom islice I forgot to update it. Thanks.

Reply:
> This will not cut it. E.g. if one sources `itertools.repeat(None)`, this goes to infinite loop.
> 
> ```python
> ...
> elif isinstance(defaults, (tuple, list)):
>     defaults = defaults[:len(items)]
> else:
>     defaults = itl.islice(defaults, len(items))
> ...
> ```
> 
> Might need alternative to `islice` not use `itertools` if it is not a dependency. And maybe some simplification to above conditionals possible.

So you would expect extra defaults to get cut-off at creation time, even for sequences? That changes things in c implementation. Just to be clear, you'd expect `itemtuplegetter([], defaults=(1, 2, 3)).defaults == ()`, right?